src/app/services/actions/query-action-creator-util.ts

```diff
@@ -30,28 +38,35 @@ export function anonymousRequest(dispatch: Function, query: IQuery) {
     'Content-Type': 'application/json',
   };
 
-  const options: IRequestOptions = { method: query.selectedVerb, headers};
+  const options: IRequestOptions = { method: query.selectedVerb, headers };
+
+  dispatch(queryRunningStatus({ isLoadingData: true }));
 
   return fetch(graphUrl, options)
     .then((resp) => {
+      dispatch(queryRunningStatus({ isLoadingData: false }));
       if (resp.ok) {
         return parseResponse(resp, respHeaders);
       }
       throw new Error('The request was not completed');
     })
     .then((json) =>
-
       dispatch(
         queryResponse({
           body: json,
           headers: respHeaders,
         }),
+        queryRunningStatus({ isLoadingData: false })
```

Comment:
By the time this `queryRunningStatus` is dispatched the value of `isLoadingData` is false because of the dispatch in `line 47`. It is safe to not dispatch `queryRunningStatus` here.

Reply:
At what point is it safer?

Reply:
I don't understand the question

Reply:
It is safe to remove line `59`